modules/quadbin/snowflake/sql/QUADBIN_TOZXY.sql

```diff
@@ -0,0 +1,63 @@
+CREATE OR REPLACE FUNCTION QUADBIN_TOZXY
+(quadbin INT)
+RETURNS OBJECT
+AS $$
+WITH
+__interleaved AS (
+  SELECT
+    BITAND(BITSHIFTRIGHT(quadbin, 59), 7) AS mode,
+    BITAND(BITSHIFTRIGHT(quadbin, 57), 3) AS extra,
+    BITAND(BITSHIFTRIGHT(quadbin, 52), 31) AS z,
+    BITSHIFTLEFT(BITAND(quadbin, TO_NUMBER('fffffffffffffff', 'XXXXXXXXXXXXXXXX')), 12) AS q
```

Comment:
I would avoid the TO_NUMBER calls and use contants; you can get the decimal values from the Postgres PR